plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/Generator.java

```diff
@@ -313,10 +318,41 @@ public String getFileHeader() {
 					result = result.replace(FILE_HEADER_VAR_USER, user);
 				}
 			}
+			if (result.contains(FILE_HEADER_VAR_VERSION)) {
+				String version = getVersion();
+				if (version != null) {
+					result = result.replace(FILE_HEADER_VAR_VERSION, version);
+				}
+			}
 		}
 		return result;
 	}
 
+	private String getVersion() {
+		// Read the exact version from the Manifest of the plugin.
+		try {
+			URL url = new URL(Plugin.INSTANCE.getBaseURL() + "META-INF/MANIFEST.MF");
+			final Manifest manifest = new Manifest(url.openStream());
+			return manifest.getMainAttributes().getValue("Bundle-Version");
+		}
+		catch (Exception e) {
+			return null;
+		}
+	}
+
+	// only needed to determine the Manifest file and its version of this plugin in standalone mode
+	private static class Plugin extends EMFPlugin {
```

Comment:
Would it be reasonable to extends DelegatingResourceLocator directly? I don't see any code that uses anything of the EMFPlugin